ceramic-kubo-rpc/src/http.rs

```diff
@@ -0,0 +1,611 @@
+//! Implements the /dag/* endpoints.
+use std::{io::Cursor, net, str::FromStr};
+
+use actix_multipart::Multipart;
+use actix_web::{
+    error,
+    http::{header::ContentType, StatusCode},
+    web, App, HttpResponse, HttpServer, Scope,
+};
+use anyhow::anyhow;
+use futures_util::StreamExt;
+use iroh_api::{Cid, IpfsPath, Multiaddr, PeerId};
+use libipld::{cbor::DagCborCodec, ipld, json::DagJsonCodec, prelude::Encode};
+use multiaddr::Protocol;
+use serde::Deserialize;
+use serde::Serialize;
+use tracing_actix_web::TracingLogger;
+
+use crate::{dag, error::Error, swarm, IpfsDep};
+
+#[derive(Clone)]
+struct AppState<T>
+where
+    T: IpfsDep,
+{
+    api: T,
+}
+
+/// Start the Kubo RPC mimic server.
+///
+/// Block until shutdown.
+/// Automatically registers shutdown listeners for interrupt and kill signals.
+/// See https://actix.rs/docs/server/#graceful-shutdown
+pub async fn serve<T, A>(api: T, addrs: A) -> std::io::Result<()>
+where
+    T: IpfsDep + Send + Clone + 'static,
+    A: net::ToSocketAddrs,
+{
+    HttpServer::new(move || {
+        App::new()
+            .wrap(TracingLogger::default())
+            .app_data(web::Data::new(AppState { api: api.clone() }))
+            .service(
+                web::scope("/api/v0")
+                    .service(dag_scope::<T>())
+                    .service(swarm_scope::<T>()),
+            )
+    })
+    .bind(addrs)?
+    .run()
+    .await
+}
+
+fn dag_scope<T>() -> Scope
```

Comment:
I'd like to see these places in their own files, with unit tests. Application level can do bind and run, while we should only expose Scopes and possibly App.

Reply:
Yeah an `http::dag` and `http::swarm` modules would make sense I can make that refactor